control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go

```diff
@@ -0,0 +1,208 @@
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "nto-metrics-client",
+							},
+							Key: "ca.crt",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	util.ApplyClusterIDLabel(&sm.Spec.Endpoints[0], clusterID)
+
+	return nil
+}
+
+func ReconcileDeployment(dep *appsv1.Deployment, params Params) error {
+	params.OwnerRef.ApplyTo(dep)
+
+	dep.Spec.Replicas = utilpointer.Int32(1)
+	dep.Spec.Selector = &metav1.LabelSelector{MatchLabels: map[string]string{"name": operatorName}}
+	if dep.Spec.Template.Annotations == nil {
+		dep.Spec.Template.Annotations = map[string]string{}
+	}
+	dep.Spec.Template.Annotations["target.workload.openshift.io/management"] = `{"effect": "PreferredDuringScheduling"}`
+	if dep.Spec.Template.Labels == nil {
+		dep.Spec.Template.Labels = map[string]string{}
+	}
+	dep.Spec.Template.Labels["name"] = operatorName
+
+	ntoArgs := []string{
+		"-v=2",
+	}
+
+	var ntoEnv []corev1.EnvVar
+
+	ntoEnv = append(ntoEnv, []corev1.EnvVar{
+		{Name: "RELEASE_VERSION", Value: params.ReleaseVersion},
+		{Name: "HYPERSHIFT", Value: "true"},
+		{Name: "MY_NAMESPACE", ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: "metadata.namespace",
+			},
+		}},
+		{Name: "WATCH_NAMESPACE", Value: "openshift-cluster-node-tuning-operator"},
+		{Name: "POD_NAME", ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: "metadata.name",
+			},
+		}},
+		{Name: "RESYNC_PERIOD", Value: "600"},
+		{Name: "CLUSTER_NODE_TUNED_IMAGE", Value: params.Images.NodeTunedContainer},
+		{Name: "KUBECONFIG", Value: "/etc/kubernetes/kubeconfig"},
+	}...)
+
+	dep.Spec.Template.Spec.Containers = []corev1.Container{{
+		Command: []string{"cluster-node-tuning-operator"},
+		Args:    ntoArgs,
+		Env:     ntoEnv,
+		Name:    operatorName,
+		Image:   params.Images.NodeTuningOperator,
+		Ports: []corev1.ContainerPort{
+			{Name: "metrics", ContainerPort: 60000},
+		},
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Resources: corev1.ResourceRequirements{Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("10m"),
+			corev1.ResourceMemory: resource.MustParse("50Mi"),
+		}},
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: "node-tuning-operator-tls", MountPath: "/etc/secrets"},
+			{Name: "trusted-ca", MountPath: "/var/run/configmaps/trusted-ca/"},
+			{Name: "hosted-kubeconfig", MountPath: "/etc/kubernetes"},
+			// TODO add {Name: "apiservice-cert", MountPath: "/apiserver.local.config/certificates"},
+
+		},
+	}}
+	dep.Spec.Template.Spec.Volumes = []corev1.Volume{
+		{Name: "node-tuning-operator-tls", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "node-tuning-operator-tls"}}},
+		{Name: "apiservice-cert", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{
+			SecretName:  "performance-addon-operator-webhook-cert",
```

Comment:
who creates this secret?

Reply:
I removed this volume as it isn't needed until we are enabling PAO / PerformanceProfile controller, which will be at a later point when the CNF team has cycles. Thanks